src/Compilers/CSharp/Portable/Binder/LocalScopeBinder.cs

```diff
@@ -381,28 +381,23 @@ private bool ReportConflictWithLocal(Symbol local, Symbol newSymbol, string name
 
             if (newSymbolKind == SymbolKind.ErrorType) return true;
 
-            if (newSymbolKind == SymbolKind.Local)
-            {
-                if (this.Locals.Contains((LocalSymbol)newSymbol) && newLocation.SourceSpan.Start >= local.Locations[0].SourceSpan.Start)
-                {
-                    // A local variable named '{0}' is already defined in this scope
-                    diagnostics.Add(ErrorCode.ERR_LocalDuplicate, newLocation, name);
-                    return true;
-                }
-            }
+            var declaredInThisScope = false;
+
+            declaredInThisScope |= newSymbolKind == SymbolKind.Local && this.Locals.Contains((LocalSymbol)newSymbol);
+            declaredInThisScope |= newSymbolKind == SymbolKind.Method && this.LocalFunctions.Contains((LocalFunctionSymbol)newSymbol);
 
-            if (newSymbolKind == SymbolKind.Method)
+            if (declaredInThisScope && newLocation.SourceSpan.Start >= local.Locations[0].SourceSpan.Start)
             {
-                if (this.LocalFunctions.Contains((LocalFunctionSymbol)newSymbol) && newLocation.SourceSpan.Start >= local.Locations[0].SourceSpan.Start)
-                {
-                    // A local variable named '{0}' is already defined in this scope
-                    diagnostics.Add(ErrorCode.ERR_LocalDuplicate, newLocation, name);
-                    return true;
-                }
+                // TODO: Message should change to something like "A {0} named '{1}' is already defined in this scope"
```

Comment:
Please file a bug vs. adding a TODO comment.  It's too easy to forget about comments, hard to forget about bugs :smile: